src/main/java/io/dgraph/DgraphClient.java

```diff
@@ -28,9 +38,301 @@
+      return new ClientOptions(host, port);
+    }
+
+    /**
+     * Sets username and password for ACL authentication.
+     *
+     * @param username The username for ACL authentication.
+     * @param password The password for ACL authentication.
+     * @return This ClientOptions instance for chaining.
+     */
+    public ClientOptions withACLCredentials(String username, String password) {
+      this.username = username;
+      this.password = password;
+      return this;
+    }
+
+    /**
+     * Sets an API key for authorization.
+     *
+     * @param apiKey The API key to use for authorization.
+     * @return This ClientOptions instance for chaining.
+     */
+    public ClientOptions withApiKey(String apiKey) {
+      this.authorizationToken = apiKey;
+      return this;
+    }
+
+    /**
+     * Sets a bearer token for authorization.
+     *
+     * @param token The bearer token to use for authorization.
+     * @return This ClientOptions instance for chaining.
+     */
+    public ClientOptions withBearerToken(String token) {
+      this.authorizationToken = "Bearer " + token;
+      return this;
+    }
+
+    /**
+     * Configures the client to use plaintext communication (no encryption).
+     *
+     * @return This ClientOptions instance for chaining.
+     */
+    public ClientOptions withPlaintext() {
+      this.channelBuilder.usePlaintext();
+      return this;
+    }
+
+    /**
+     * Configures the client to use TLS but skip certificate validation.
+     * Be aware this disables certificate validation and significantly reduces the
+     * security of TLS. This mode should only be used in non-production
+     * (e.g., testing or development) environments.
+     *
+     * @return A new ClientOptions instance with TLS but without certificate validation.
+     * @throws SSLException If there's an error configuring the SSL context.
+     */
+    public ClientOptions withTLSSkipVerify() throws SSLException {
+      SslContext sslContext = GrpcSslContexts.forClient()
+          .trustManager(InsecureTrustManagerFactory.INSTANCE)
+          .build();
+
+      // Create a new options object with the same credentials
+      ClientOptions newOptions = new ClientOptions(host, port) {
+        @Override
+        public DgraphGrpc.DgraphStub createStub() {
+          NettyChannelBuilder nettyBuilder = NettyChannelBuilder.forAddress(host, port);
+          nettyBuilder.sslContext(sslContext);
+          return DgraphGrpc.newStub(nettyBuilder.build());
+        }
+      };
+
+      // Copy over the auth settings
+      newOptions.username = this.username;
+      newOptions.password = this.password;
+      newOptions.authorizationToken = this.authorizationToken;
+      return newOptions;
+    }
+
+    /**
+     * Configures the client to use TLS with certificate validation.
+     *
+     * @return This ClientOptions instance for chaining.
+     */
+    public ClientOptions withTLS() {
```

Comment:
which function is similar to in dgo for this function?

Reply:
WithSystemCertPool